deltacat/utils/ray_utils/RetryHandler/retry.py

```diff
@@ -0,0 +1,63 @@
+import ray
+import time
+import logging
+from typing import List, Callable
+from ray.types import ObjectRef
+from RetryExceptions.retryable_exception import RetryableException
+from RetryExceptions.non_retryable_exception import NonRetryableException
+from RetryExceptions.TaskInfoObject import TaskInfoObject
+
+#inputs: task_callable, task_input, ray_remote_task_options, exception_retry_strategy_configs
+#include a seperate class for errors: break down into retryable and non-retryable
+#seperate class to put info in a way that the retry class can handle: ray retry task info
+
+#This is what specifically retries a single task
+@ray.remote
+def submit_single_task(taskObj: TaskInfoObject) -> Any:
+    try:
+        taskObj.attempt_count += 1
+        curr_attempt = taskObj.attempt_count
+        return tackObj.task_callable(taskObj.task_input)
+    except (Exception) as exception:
+        # if exception is detected we want to figure out how to handle it
+        #pass to a new method that handles exception strategy
+        #retry_strat = ...exception_retry_strategy_configs
+        retry_config = get_retry_strategy() #need to come up with fields needed for this
+        if retry_config is not None:
+            return the exception that retry_config detected
+
+
+
+class RetryHandler:
```

Comment:
This does not just perform retry right? It will also submit and run multiple tasks ensuring parallelism.